CI: combine analyzer, s3, dbreplicated into one job

maxknv opened the pull request:
#job_Stateless_tests_release_analyzer_s3_DatabaseReplicated
 #job_style_check
 #no_merge_commit

<!---
A technical comment, you are free to remove or leave it as it is when PR is created
The following categories are used in the next scripts, update them accordingly
utils/changelog/changelog.py
tests/ci/cancel_and_rerun_workflow_lambda/app.py
-->
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)


### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
...

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

<!---
Directly edit documentation source files in the "docs" folder with the same pull-request as code changes

or

Add a user-readable short description of the changes that should be added to docs.clickhouse.com below.

At a minimum, the following information should be added (but add more as needed).
- Motivation: Why is this function, table engine, etc. useful to ClickHouse users?

- Parameters: If the feature being added takes arguments, options or is influenced by settings, please list them below with a brief explanation.

- Example use: A query or command.
-->


> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

maxknv committed:
CI: combine analyzer s3 dbreplicated into one job

 #job_Stateless_tests_release_analyzer_s3_DatabaseReplicated
 #job_style_check
 #no_merge_commit

robot-ch-test-poll commented:
<!-- automatic status comment for PR #60224 from ClickHouse/ClickHouse:ci_combine_analyzer_s3_replicated_tests -->
*This is an automated comment for commit 9fb55f1fa45e70819977a4b3f84c151fdbf03e52 with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/60224/9fb55f1fa45e70819977a4b3f84c151fdbf03e52/ci_running.html) to open a full report in a separate page

<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/60224/9fb55f1fa45e70819977a4b3f84c151fdbf03e52/ci_running.html">⏳ pending</a></td></tr>
<tbody>
</table>

maxknv committed:
WA: 01023 test suppressed for replicated db

 #job_Stateless_tests_release_analyzer_s3_DatabaseReplicated #batch_2
 #job_style_check
 #no_merge_commit
